packages/web-app-files/tests/unit/components/SideBar/Links/PublicLinks/LinkEdit.spec.js

```diff
@@ -0,0 +1,618 @@
+              createStore({
+                linkInEdit: { name: 'Public Link' },
+                publicLinkCapabilities: getLinkCapabilities({
+                  passwordEnforcedFor: {
+                    read_only: 1
+                  },
+                  enabledExpireDate: null,
+                  days: null
+                })
+              }),
+              { saving: false }
+            )
+            const linkCreateButton = wrapper.find(selectors.linkCreateButton)
+            expect(linkCreateButton.exists()).toBeTruthy()
+            expect(linkCreateButton.attributes('disabled')).toBeFalsy()
+          })
+          it('should enable link create button if form is valid', async () => {
+            const addLinkSpy = jest.spyOn(mapActions, 'addLink')
+            const wrapper = getMountedWrapper(
+              createStore({
+                linkInEdit: { name: 'Public Link', hasPassword: true },
+                publicLinkCapabilities: getLinkCapabilities({ enforcedExpireDate: true })
+              }),
+              { saving: false }
+            )
+            const linkCreateButton = wrapper.find(selectors.linkCreateButton)
+            expect(linkCreateButton.attributes('disabled')).toBeFalsy()
+
+            await linkCreateButton.trigger('click')
+            expect(wrapper.vm.saving).toBeTruthy()
+            expect(addLinkSpy).toHaveBeenCalledTimes(1)
+          })
+        })
+        describe('when existing link is being updated', () => {
+          it('should show link save button', () => {
+            const wrapper = getShallowMountedWrapper(
+              createStore({
+                linkInEdit: { id: 1224, name: 'Public Link' }
+              }),
+              { saving: false }
+            )
+
+            const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+            expect(linkSaveButton.exists()).toBeTruthy()
+          })
+          it('should set link save button as disabled if form is not valid', () => {
+            const wrapper = getShallowMountedWrapper(
+              createStore({
+                linkInEdit: { id: 1224, name: 'Public Link' },
+                // invalid password field
+                // by default viewer role is selected with permission 1 (viewer)
+                // combined with enforced for password is enforce for link form
+                // since link password is enforced and password is not set, form will be invalid
+                publicLinkCapabilities: getLinkCapabilities({
+                  passwordEnforcedFor: {
+                    read_only: '1'
+                  }
+                })
+              }),
+              { saving: false }
+            )
+            const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+
+            expect(linkSaveButton.attributes('disabled')).toBeTruthy()
+          })
+          it('should set link save button as disabled if form does not have any changes', () => {
```

Comment:
looks like the test before, so how do we know that the button is disabled because of the date not been changed or because of an invalid form?

Reply:
test fixed